components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java

```diff
@@ -3975,9 +3984,21 @@ public String addIdPWithResourceId(IdentityProvider identityProvider, int tenant
             throw new IdentityProviderManagementException("An error occurred while processing content stream.", e);
         } catch (SQLException e) {
             IdentityDatabaseUtil.rollbackTransaction(dbConnection);
+            /* Since only one federated authenticator per newly creating IDP is allowed, the custom IDPs will always
+            have a single federated authenticator. For older IDPs, executing this 'if' code block is unnecessary. */
+            if (identityProvider.getFederatedAuthenticatorConfigs().length == 1) {
```

Comment:
Rather than just relying on the available authenticators can't we rely on defined by property ?

Reply:
We have to check whether IDP has at least on fed authenticator, as there can be idp with empty fed auth list. The defined by property is for fed authenticator, for idp there is no such property.

Reply:
ok make sense